dev/com.ibm.ws.jdbc/test/com/ibm/ws/jdbc/internal/InternalTest.java

```diff
@@ -165,4 +166,73 @@ public void testInferClassNamesFromPID() {
+
+        //URL with no descriptor beyond vendor, no properties
+        assertEquals("Test2", "jdbc:vendor:****", 
+                     PropertyService.hidePasswordsInURL("jdbc:vendor:username/pass123!@localhost:1521:oracle"));
+
+        //URL with third piece of information, with = symbol in password
+//       assertEquals("Test3", "jdbc:oracle:thin:****", 
+//                    PropertyService.hidePasswordsInURL("jdbc:oracle:thin:username/pass123=!@localhost:1521:oracle"));
+
+        //Test with password in URL, with qualifier following, using semicolon
+        assertEquals("Test4", "jdbc:sqlserver:****;user=username;password=****;", 
+                     PropertyService.hidePasswordsInURL("jdbc:sqlserver:!@pass://localhost;user=username;password=ab9&*&^*#(*;"));
+
+        //Test with password at very end of URL, using semicolon
+        assertEquals("Test5", "jdbc:sqlserver:****;user=username;password=****", 
+                     PropertyService.hidePasswordsInURL("jdbc:sqlserver://localhost/;user=username;password=ab9&*&^*#(*"));
+
+        //Test with password using ? then & pattern
+        assertEquals("Test6", "jdbc:postgresql:****?user=alex&password=****&ssl=true&****",
+                     PropertyService.hidePasswordsInURL("jdbc:postgresql://localhost/test?user=alex&password=Minn;esota;55901&ssl=true&blank"));
+
+        //Test with password listed first using ? then & pattern
+        assertEquals("Test7", "jdbc:postgresql:****?password=****&foo=&user=alex&blank=",
+                     PropertyService.hidePasswordsInURL("jdbc:postgresql://localhost/test?password=Minn;esota;55901&foo=&user=alex&blank="));
+
+        //Test with two passwords in url
+        assertEquals("Test8", "jdbc:sqlserver:****;user=username;password=****;foo=bar;password=****;", 
+                     PropertyService.hidePasswordsInURL("jdbc:sqlserver://localhost;user=username;password=ab9&*&^*#(*;foo=bar;password=Rochester;"));
+
+        //Test with PWD at the end of the url (with a token in the password)
+        assertEquals("Test9", "jdbc:vendor:****;UI_D=te_st;PWD=****;****", 
+                     PropertyService.hidePasswordsInURL("jdbc:vendor://localhost;UI_D=te_st;PWD=Zom;bies"));
+
+        //Test with PWD and property name containing password
+        assertEquals("Test10", "jdbc:vendor:****;UID=test;PWD=****;****;APasswordValue=****", 
+                     PropertyService.hidePasswordsInURL("jdbc:vendor://localhost;UID=test;PWD=Rochester;Password55901!@#;APasswordValue=IBM"));
+
+        //Test with multiple and empty passwords specified, using colon then semicolon pattern
+        assertEquals("Test11", "jdbc:db2:****:user=dbadm;password=****;password=", 
+                     PropertyService.hidePasswordsInURL("jdbc:db2://sysmvs1.stl.ibm.com:5021/STLEC1:user=dbadm;password=dbadm;password="));
+
+//        assertEquals("Test12", "jdbc:mysql:****", 
```

Comment:
commented out tests without any explanation.  Did you forget to enable them?  Or are there plans to get them working in a subsequent pull?